src/v/cluster/rm_stm.cc

```diff
@@ -2650,4 +2650,16 @@ ss::future<> rm_stm::clear_old_tx_pids() {
       });
 }
 
+ss::future<> rm_stm::clear_old_idempotent_pids() {
+    while (_log_state.idempotent_pids_order.size() > 100500) {
```

Comment:
as I understand when we load the snapshot the list is empty so there is still a chance that it will always grow

Reply:
No
We added pids inside this map on loading shapshot